bittensor/_neuron/text/advanced_server/run.py

```diff
@@ -252,11 +252,10 @@ def time_check():
         data = next(dataset)
 
         # --- creating our chain weights
-        chain_weights = torch.zeros(metagraph.n)
+        # --- query the chain for the most current number of peers on the network
+        chain_weights = torch.zeros(subtensor.n)
         uid = nn.uid
-       
-        if uid < len(chain_weights):
-            chain_weights[uid] = 1 
+        chain_weights[uid] = 1 
```

Comment:
We need a better fix than this, because people's servers have been hanging because the UID doesn't exist in metagraph yet.

Reply:
We now query the chain for the number of peers on the network. If they are registered, their uid will be included as part of the set.